src/nlsolve/jfnk.jl

```diff
@@ -0,0 +1,56 @@
+"""
+    SimpleJFNK()
+A low overhead Jacobian-free Newton-Krylov method. This method internally uses `GMRES` to
+avoid computing the Jacobian Matrix.
+!!! warning
+    JFNK doesn't work well without preconditioning, which is currently not supported. We
```

Comment:
```suggestion
    JFNK may not work well without preconditioning, which is currently not supported. We
```

There's lots of precedence on this kind of implementation, so I wouldn't say it won't be useful.